torch/_inductor/codegen/cpp.py

```diff
@@ -3714,26 +3702,52 @@ def is_lowp_fp_scheduler(self, scheduler_node: SchedulerNode):
 
     def legalize_lowp_fp_dtype_loopbody(self, loop_body: LoopBody):
         def add_to_dtype(sub_graph: torch.fx.Graph):
-            def is_lowp_fp_load(node: torch.fx.Node):
-                if node.target not in ["load"]:
-                    return False
-                assert len(node.args) == 3
-                load_dtype = V.graph.get_dtype(node.args[1])  # type: ignore[arg-type]
-                return load_dtype in DTYPE_LOWP_FP
+            def get_input_dtype(node: torch.fx.Node):
+                """Get input dtype for nodes that may consumes lowp fp dt; True for ops that can accept any dt of input"""
+                if node.target == "store":
+                    return V.graph.get_dtype(node.args[1])
+                elif node.target == "to_dtype_bitcast":
+                    return node.args[-1]
+                elif node.target == "to_dtype":
+                    src_dtype = node.kwargs.get("src_dtype", None)
+                    return True if src_dtype is None else src_dtype
```

Comment:
Why `src_dtype is None` means this `to_dtype` node accept any dt of input?

Reply:
It seems that most use cases of `to_dtype` are just don't pass the `src_dtype`. Specifically, if a `to_dtype` follows an node that generates lowp fp data, it should be able to accept it and convert it to the `dtype` in args without accuracy risk. By contrast, most nodes, cannot accept lowp fp input for accuracy concerns now.
![image](https://github.com/user-attachments/assets/752a7501-6e2b-410c-b4df-c6bf2d022e98)